src/devices/Arduino/ArduinoAnalogInputPin.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Analog;
+using System.Device.Gpio;
+using System.Text;
+
+namespace Iot.Device.Arduino
+{
+    internal class ArduinoAnalogInputPin : AnalogInputPin
+    {
+        private readonly SupportedPinConfiguration _configuration;
+        private int _autoReportingReferenceCount;
+        private ArduinoBoard _board;
+
+        public ArduinoAnalogInputPin(ArduinoBoard board, AnalogController controller, SupportedPinConfiguration configuration,
+            int pinNumber, double voltageReference)
+            : base(controller, pinNumber, voltageReference)
+        {
+            _board = board;
+            _configuration = configuration;
+        }
+
+        public override void EnableAnalogValueChangedEvent(GpioController? masterController, int masterPin)
```

Comment:
I'm a bit confused on what this does and when it's used. Why not have event as part of this class?

Reply:
This event is the link to the underlying transport connection and therefore binding-specific. The public event that you're looking for is in the base class. Forwarding takes place further down in `FirmataOnAnalogPinValueUpdated`. The base class has no direct knowledge about the source of the event.